tex/latex/biblatex/blx-case-expl3.sty

```diff
@@ -223,12 +223,56 @@
     }
 }
 
-
-\cs_new:Npn \biblatex_text_sentencecase:nn #1 #2
+% sentence casing is a bit more involved due to expl3 changes
+% <https://github.com/latex3/latex3/issues/1232>
+% <https://github.com/latex3/latex3/pull/1240>
+% <https://github.com/latex3/latex3/pull/1247>
+% In old versions we use \text_titlecase:n
+% (counter-intuitive given its name).
+% In new versions that is deprecated and has to be emulated by
+% a combination of \text_titlecase_first:n { \text_lowercase:n {#1} }.
+% This combination would not work in old versions due to a small
+% oversight, for which we add a hotfix.
+% We detect new vs. old by the presence of \text_titlecase_all:n,
+% which was added for the new version.
+\cs_if_exist:NTF \text_titlecase_all:n
   {
-    \__biblatex_text_makeconverter:nnn {titlecase} {#1} {#2}
+    \cs_new:Npn \biblatex_text_sentencecase:nn #1 #2
+      {
+        \group_begin:
+          \bool_if_exist:NTF \l_text_titlecase_check_letter_bool
```

Comment:
This test could be skipped: the `bool` predates the introduction of `\text_titlecase_all:n`.

Reply:
Ahh, good point. Thanks. Done in https://github.com/plk/biblatex/pull/1310/commits/110798a4c00e7d72386cfd035adf2de5ea2e3b44.